app/src/main/java/com/cmput301f20t21/bookfriends/ui/profile/ProfileViewModel.java

```diff
@@ -1,8 +1,78 @@
 package com.cmput301f20t21.bookfriends.ui.profile;
 
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 
+import com.cmput301f20t21.bookfriends.callbacks.OnFailCallback;
+import com.cmput301f20t21.bookfriends.callbacks.OnSuccessCallbackWithMessage;
+import com.cmput301f20t21.bookfriends.entities.User;
+import com.cmput301f20t21.bookfriends.services.UserService;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.util.ArrayList;
+
 public class ProfileViewModel extends ViewModel {
-    // TODO: Implement the ViewModel
+    private final UserService userService;
+    private final MutableLiveData<ArrayList<User>> searchedUsers = new MutableLiveData<>(new ArrayList<>());
+
+    public ProfileViewModel() {
+        userService = UserService.getInstance();
+    }
+
+    // the data stream output for searched list views
+    public LiveData<ArrayList<User>> getSearchedUsers() {
+        return searchedUsers;
+    }
+
+    // the update api for views
+    public void updateSearchQuery(String query) {
+        // start a search routine, update results
+        userService.getByUsernameStartWith(query).addOnCompleteListener(usernameTask -> {
+            if (usernameTask.isSuccessful()) {
+                if (!usernameTask.getResult().isEmpty()) {
+                    ArrayList<User> users = new ArrayList<>();
+                    for (QueryDocumentSnapshot document: usernameTask.getResult()) {
+                        users.add(new User(
+                                document.getId(),
+                                document.get("username").toString(),
+                                "",
+                                document.get("email").toString(),
+                                ""
```

Comment:
In the requirement, we only need to display user name and email. I'm thinking of removing phone number and display name to simplify everything.

Reply:
sounds good to me. should we do that in another PR?